internal/imgbuild/jobmanager.go

```diff
@@ -0,0 +1,165 @@
+package imgbuild
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/module"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type JobType string
+
+const (
+	JobTypeBuild JobType = "build"
+	JobTypeSign  JobType = "sign"
+)
+
+type BaseJobManager struct {
+	client    client.Client
+	jobHelper JobHelper
+	jobType   JobType
+	maker     JobMaker
+	registry  registry.Registry
+}
+
+func (bjm *BaseJobManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
+	jobs, err := bjm.jobHelper.GetModuleJobs(ctx, modName, namespace, bjm.jobType, owner)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get build jobs for module %s: %v", modName, err)
+	}
+
+	deleteNames := make([]string, 0, len(jobs))
+	for _, job := range jobs {
+		if job.Status.Succeeded == 1 {
+			err = bjm.jobHelper.DeleteJob(ctx, &job)
+			if err != nil {
+				return nil, fmt.Errorf("failed to delete build job %s: %v", job.Name, err)
+			}
+			deleteNames = append(deleteNames, job.Name)
+		}
+	}
+	return deleteNames, nil
+}
+
+func (bjm *BaseJobManager) Sync(ctx context.Context, mld *api.ModuleLoaderData, pushImage bool, owner metav1.Object) (Status, error) {
+	logger := log.FromContext(ctx).WithValues("type", bjm.jobType)
```

Comment:
lets add at least Name from mld to the WithValues, so that we can find in the log which module are we talking about here